src/server/game/AI/CreatureAI.cpp

```diff
@@ -28,11 +28,13 @@
 #include "CreatureTextMgr.h"
 
 //Disable CreatureAI when charmed
-void CreatureAI::OnCharmed(bool /*apply*/)
+void CreatureAI::OnCharmed(bool apply)
 {
-    //me->IsAIEnabled = !apply;*/
-    me->NeedChangeAI = true;
-    me->IsAIEnabled = false;
+    if (apply)
```

Comment:
You completely missed the point of the PR. The point is that the original creatureAI will now also get an OnCharmed invocation with apply=false, thus meaning the base CreatureAI has to check whether this flag is true or false before initiating the AI change...